safenode/src/bin/kadnode.rs

```diff
@@ -1,233 +1,178 @@
-        self.kademlia.get_closest_peers(addr.to_vec());
-    }
-}
-
-#[allow(clippy::large_enum_variant)]
-enum SafeNetBehaviour {
-    Kademlia(KademliaEvent),
-    Mdns(mdns::Event),
-}
-
-impl From<KademliaEvent> for SafeNetBehaviour {
-    fn from(event: KademliaEvent) -> Self {
-        SafeNetBehaviour::Kademlia(event)
-    }
-}
-
-impl From<mdns::Event> for SafeNetBehaviour {
-    fn from(event: mdns::Event) -> Self {
-        SafeNetBehaviour::Mdns(event)
-    }
-}
-
-#[derive(Debug)]
-enum SwarmCmd {
-    Search(XorName),
-}
-
-/// Channel to send Cmds to the swarm
-type CmdChannel = tokio::sync::mpsc::Sender<SwarmCmd>;
-
-fn run_swarm() -> CmdChannel {
-    let (sender, mut receiver) = tokio::sync::mpsc::channel::<SwarmCmd>(1);
-
-    let _handle: tokio::task::JoinHandle<Result<(), Error>> = tokio::spawn(async move {
-        debug!("Starting swarm");
-        // Create a random key for ourselves.
-        let keypair = identity::Keypair::generate_ed25519();
-        let local_peer_id = PeerId::from(keypair.public());
-
-        // QUIC configuration
-        let quic_config = libp2p_quic::Config::new(&keypair);
-        let transport = libp2p_quic::async_std::Transport::new(quic_config);
-        let transport = transport
-            .map(|(peer_id, muxer), _| (peer_id, StreamMuxerBox::new(muxer)))
-            .boxed();
-
-        // Create a Kademlia instance and connect to the network address.
-        // Create a swarm to manage peers and events.
-        let mut swarm = {
-            // Create a Kademlia behaviour.
-            let mut cfg = KademliaConfig::default();
-            cfg.set_query_timeout(Duration::from_secs(5 * 60));
-            let store = MemoryStore::new(local_peer_id);
-            let kademlia = Kademlia::new(local_peer_id, store);
-            let mdns = mdns::async_io::Behaviour::new(mdns::Config::default(), local_peer_id)?;
-            let behaviour = MyBehaviour { kademlia, mdns };
-
-            let mut swarm =
-                SwarmBuilder::with_async_std_executor(transport, behaviour, local_peer_id).build();
-
-            // // Listen on all interfaces and whatever port the OS assigns.
-            let addr = "/ip4/0.0.0.0/udp/0/quic-v1".parse().expect("addr okay");
-            swarm.listen_on(addr).expect("listening failed");
-
-            swarm
-        };
-
-        let net_info = swarm.network_info();
-
-        debug!("network info: {net_info:?}");
-        // Kick it off.
+#[async_std::main]
+async fn main() -> Result<()> {
+    let opt = Opt::parse();
+    let _log_appender_guard = init_node_logging(&opt.log_dir)?;
+
+    let (mut network_api, mut network_events, network_event_loop) = EventLoop::new()?;
+    let temp_dir = TempDir::new()?;
+    let storage = DataStorage::new(&temp_dir);
+
+    // Spawn the network task for it to run in the background.
+    spawn(network_event_loop.run());
+
+    // Todo: Implement node bootstrapping to connect to peers from outside local network
```

Comment:
let's pull this out into a bootstrap func or something, and mark as unused for now